File: go/monitor.go
package main

import (
	"context"
	"fmt"
	"os"

	"go.mongodb.org/mongo-driver/v2/event"
)

func makeMonitor() *event.CommandMonitor {
	if os.Getenv("PRINT_MONGODB_COMMANDS") == "" {
		return nil
	}
	return &event.CommandMonitor{
		Started: func(_ context.Context, e *event.CommandStartedEvent) {
			fmt.Printf(">> %s %s\n", e.CommandName, e.Command)
		},
		Succeeded: func(_ context.Context, e *event.CommandSucceededEvent) {
			fmt.Printf("<< %s %s\n", e.CommandName, e.Reply)
		},
	}
}

File: rust/src/monitor.rs
use mongodb::event::{command::CommandEvent, EventHandler};

pub fn make_handler() -> Option<EventHandler<CommandEvent>> {
    if std::env::var("PRINT_MONGODB_COMMANDS").is_err() {
        return None;
    }
    Some(EventHandler::callback(|event| match event {
        CommandEvent::Started(e) => println!(">> {} {}", e.command_name, e.command),
        CommandEvent::Succeeded(e) => println!("<< {} {}", e.command_name, e.reply),
        _ => {}
    }))
}
